### ISSUE TITLE
batch SimilarityTransform

### ISSUE THREAD

**Issue by muqishan** [deepinsight/insightface#2328]
The following code implements face alignment using functions from the `skimage` library. In cases where there are a small number of faces, using this function for face alignment can yield satisfactory results. However, when dealing with a large collection of faces, I'm looking for a method to calculate similarity transformation matrices in batches. I have already achieved batch face alignment using similarity transformation matrices of shape [b, 2, 3] with the `kornia` library. However, I'm struggling to find a way to calculate similarity transformation matrices in batches while maintaining consistent results with the computation performed by `skimage`. Furthermore, I'm hoping to accomplish this using the PyTorch framework. I have attempted to replicate the computation of similarity transformation matrices from `skimage` using PyTorch, but the results do not match. This discrepancy could impact the accuracy of subsequent face recognition tasks. Has anyone successfully implemented this? Any help would be greatly appreciated.

```
   tform = trans.SimilarityTransform()
    tform.estimate(src,dst)
    M = tform.params[0:2,:]
```
can stack the results of `trans.SimilarityTransform()` into a [n, 2, 3] matrix, which can then be used with `kornia.geometry.transform.warp_affine` along with the corresponding images to perform batch alignment tasks.

**Comment by conansherry** [comment]
example code below
`import cv2
import numpy as np
import torch

from backbones import get_model
from face_align import norm_crop2, estimate_norm

from insightface.app import FaceAnalysis
import torch.nn.functional as F

import kornia

@torch.no_grad()
def inference(net, img):
    img = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
    img = np.transpose(img, (2, 0, 1))
    img = torch.from_numpy(img).unsqueeze(0).float().cuda()
    img.div_(255).sub_(0.5).div_(0.5)
    feat = net(img)
    feat = F.normalize(feat, p=2, dim=1)
    feat = feat.cpu().numpy()
    return feat

if __name__ == "__main__":
    net = get_model('r100', fp16=False)
    weights = torch.load('./weights/model.pt')
    info = net.load_state_dict(weights)
    print('net', info)
    net.eval()
    net.cuda()

    app = FaceAnalysis(providers=['CUDAExecutionProvider', 'CPUExecutionProvider'], allowed_modules=['detection', 'landmark_2d_106'])
    app.prepare(ctx_id=0, det_size=(640, 640))

    # get img1
    img1 = cv2.imread("./images/src1.png")
    img2 = cv2.imread("./images/src2.png")

    # get face1
    faces1 = app.get(img1)
    # get face2
    faces2 = app.get(img2)

    faces1_kpt5 = faces1[0].kps
    faces2_kpt5 = faces2[0].kps

    # img1 to tensor
    img = cv2.cvtColor(img1, cv2.COLOR_BGR2RGB)
    img = np.transpose(img, (2, 0, 1))
    img = torch.from_numpy(img).unsqueeze(0).float().cuda()
    img.div_(255).sub_(0.5).div_(0.5)
    img1_tensor = img
    # img2 to tensor
    img = cv2.cvtColor(img2, cv2.COLOR_BGR2RGB)
    img = np.transpose(img, (2, 0, 1))
    img = torch.from_numpy(img).unsqueeze(0).float().cuda()
    img.div_(255).sub_(0.5).div_(0.5)
    img2_tensor = img

    # faces1_kpt5 to tensor
    faces1_kpt5_tensor = torch.from_numpy(faces1_kpt5).unsqueeze(0).float().cuda()
    # faces2_kpt5 to tensor
    faces2_kpt5_tensor = torch.from_numpy(faces2_kpt5).unsqueeze(0).float().cuda()

    # estimate matrix
    M1 = estimate_norm(faces1_kpt5)
    M2 = estimate_norm(faces2_kpt5)
    # pad [0, 0, 1] to M
    M1 = np.row_stack((M1, np.array([0, 0, 1])))
    M2 = np.row_stack((M2, np.array([0, 0, 1])))

    # M to tensor
    M1_tensor = torch.from_numpy(M1).unsqueeze(0).float().cuda()
    M2_tensor = torch.from_numpy(M2).unsqueeze(0).float().cuda()

    print('img1_tensor.shape', img1_tensor.shape)
    print('img2_tensor.shape', img2_tensor.shape)
    print('faces1_kpt5_tensor.shape', faces1_kpt5_tensor.shape)
    print('faces2_kpt5_tensor.shape', faces2_kpt5_tensor.shape)
    print('M1_tensor.shape', M1_tensor.shape)
    print('M2_tensor.shape', M2_tensor.shape)

    stacked_img_tensor = torch.cat((img1_tensor, img2_tensor), dim=0)
    stacked_M_tensor = torch.cat((M1_tensor, M2_tensor), dim=0)
    # warped = cv2.warpAffine(img, M, (image_size, image_size), borderValue=0.0)
    # use kornia to warp
    warped = kornia.geometry.warp_perspective(stacked_img_tensor, stacked_M_tensor, dsize=(112, 112))

    # wraped has 2 images. split to 2 images
    warped1 = warped[0]
    warped2 = warped[1]

    # wraped1 is tensor. convert to numpy
    warped1 = warped1.squeeze(0).permute(1, 2, 0).cpu().numpy()
    wraped1 = warped1 * 0.5 + 0.5
    wraped1 = wraped1 * 255
    wraped1 = wraped1.astype(np.uint8)
    wraped1 = cv2.cvtColor(wraped1, cv2.COLOR_RGB2BGR)
    cv2.imshow('warped1', wraped1)

    # wraped2 is tensor. convert to numpy
    warped2 = warped2.squeeze(0).permute(1, 2, 0).cpu().numpy()
    wraped2 = warped2 * 0.5 + 0.5
    wraped2 = wraped2 * 255
    wraped2 = wraped2.astype(np.uint8)
    wraped2 = cv2.cvtColor(wraped2, cv2.COLOR_RGB2BGR)
    cv2.imshow('warped2', wraped2)

    cv2.waitKey(0)


`
